src/mono/mono/metadata/icall.c

```diff
@@ -6030,9 +6033,10 @@ ves_icall_System_Reflection_Assembly_InternalGetAssemblyName (MonoStringHandle f
 
 	if (!image){
 		if (status == MONO_IMAGE_IMAGE_INVALID)
-			mono_error_set_bad_image_by_name (error, filename, "Invalid Image");
-		else
-			mono_error_set_file_not_found (error, filename, "%s", "");
+			mono_error_set_bad_image_by_name (error, filename, "Invalid Image: %s", filename);
+		else {
```

Comment:
Nit: no reason to add the curly braces here, Mono style is to omit them for one-line bodies.